components/docker/docker.go

```diff
@@ -0,0 +1,170 @@
+package docker
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/namer"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/DataDog/test-infra-definitions/components/command"
+	remoteComp "github.com/DataDog/test-infra-definitions/components/remote"
+
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	composeVersion = "v2.12.2"
+	defaultTimeout = 300
+)
+
+type Manager struct {
+	namer namer.Namer
+	host  *remoteComp.Host
+
+	opts []pulumi.ResourceOption
+}
+
+func NewManager(e config.CommonEnvironment, host *remoteComp.Host, installDocker bool, opts ...pulumi.ResourceOption) (*Manager, pulumi.Resource, error) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
So manager is no more lazy, right ?

Reply:
That is correct, if you instantiate a `DockerManager` without calling it, it will still install Docker. Not sure it's a valid use case though.

Reply:
Reading it from `datadog-agent` we should have `NewManagerWithoutDockerInstall` and `NewManagerWithDockerInstall`. Can be done in a follow up

Reply:
Isn't it the goal of the boolean? I think that we should move the `Docker` install to be conditional on presence of `docker` CLI, it would improve the experience without being tied to a specific OS.
It's for follow-up though.

Reply:
The boolean makes it harder to read the purpose, while implementing the feature. In other words, using it from `datadog-agent` it is more explicit to use 

```go
docker.NewManagerWithoutDockerInstall()
``` 

rather than

```go
docker.NewManager(false)
```